runtime/shared_common/shrinit.cpp

```diff
@@ -3177,6 +3182,47 @@ j9shr_isBCIEnabled(J9JavaVM *vm)
 	return (0 != (vm->sharedClassConfig->runtimeFlags & J9SHR_RUNTIMEFLAG_ENABLE_BCI));
 }
 
+static UDATA
+romToRamGetRomAddress(void *item)
+{
+	RomToRamQueryEntry *queryEntry = (RomToRamQueryEntry *)item;
+	UDATA romAddress = (UDATA)(queryEntry->romClass);
+	if (J9_ARE_ALL_BITS_SET(romAddress, ROM_TO_RAM_QUERY_TAG)) {
+		return romAddress & ~ROM_TO_RAM_QUERY_TAG;
+	} else {
+		RomToRamEntry *entry = (RomToRamEntry *)item;
+		return (UDATA)(entry->ramClass->romClass);
+	}
+}
+
+/* THREADING: Must be protected by J9SharedClassConfig.romToRamHashTableMutex */
+static UDATA
+romToRamHashFn(void *item, void *userData)
+{
+	return romToRamGetRomAddress(item);
+}
+
+/* THREADING: Must be protected by J9SharedClassConfig.romToRamHashTableMutex */
+static UDATA
+romToRamEqualFn(void *left, void *right, void *userData)
+{
+	return (romToRamGetRomAddress(left) == romToRamGetRomAddress(right));
+}
+
+static void
+romToRamRemoveEntry(J9HookInterface **hookInterface, UDATA eventNum, void *voidData, void *userData)
```

Comment:
The code related to class unload should be inside` #if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)`.